src/network/networkd-network-gperf.gperf

```diff
@@ -256,6 +256,10 @@ IPv6RoutePrefix.Route,                       config_parse_route_prefix,
 IPv6RoutePrefix.LifetimeSec,                 config_parse_route_prefix_lifetime,                       0,                             0
 CAN.BitRate,                                 config_parse_si_uint64,                                   0,                             offsetof(Network, can_bitrate)
 CAN.SamplePoint,                             config_parse_permille,                                    0,                             offsetof(Network, can_sample_point)
+CAN.DataBitRate,                             config_parse_si_uint64,                                   0,                             offsetof(Network, can_data_bitrate)
```

Comment:
please introduce a proper parser for this, i.e. that checks the specified value for the actual bit rate range that is accepted here. I mean, 0 is not accepted as i understand (if it was, then we couldn't use it as a "does not apply" marker above, could we), and it's limited to the 32bit range too, no?

Reply:
The value 0 does indeed not make sense here, I guess right now, the existing CAN code just uses this value to check whether the setting is present or not. How should I go about this? Introduce a `config_parse_can_bitrate`, base it upon the `config_parse_si_uint32/64` with some additional checks, or simply use `config_parse_si_uint32/64` and do the checks where they currently are?

As for the an acceptable range, I don't think we should set a specific limit beyond `[U]INT32_MAX`. There are a few commonly used CAN bitrates in practice, but AFAIK there are no limits or presets inherent in the protocol. Just the physical limits of most controllers/transceivers (< 1MBIT for CAN, 8 MBIT for CAN-FD as far as I'm aware, ~10MBIT for the upcoming CAN-XL standard).

Reply:
provide your own config_parse_xyz function, yes. Depending on what is nicer/shorter, it might just wrap config_parse_si_uint32, or just do what it does itself...

Reply:
I addressed this with a `config_parser_can_bitrate` function.